demos/python_demos/bert_question_answering/question_answering_demo.py

```diff
@@ -0,0 +1,338 @@
+            tokens_id.append(tok)
+            tokens_se.append((s, e))
+    log.info("Size: {} tokens".format(len(tokens_id)))
+    return tokens_id, tokens_se
+
+
+# return entire sentence as start-end positions for a given answer (within the sentence).
+def find_sentence_range(context, s, e):
+    # find start of sentence
+    for c_s in range(s, max(-1, s - 200), -1):
+        if context[c_s] in "\n\.":
+            c_s += 1
+            break
+
+    # find end of sentence
+    for c_e in range(max(0, e - 1), min(len(context), s + 200), +1):
+        if context[c_e] in "\n\.":
+            break
+
+    return c_s, c_e
+
+
+# return context as one big string by given input arguments
+def get_context(args):
+    log.info("Get context from {}".format(args.input))
+    response = requests.get(args.input)
+    if response.status_code == 404:
+        raise Exception('URL is invalid')
+    html = bs4.BeautifulSoup(response.text, 'html.parser')
+    heads = html.select("#firstHeading")
+    title = heads[0].text if heads else "no title"
+    paragraphs = html.select("p")
+    log.info("Article title: '{}'".format(title))
+    context = '\n'.join([par.text for par in paragraphs])
+    log.info("Size: {} chars".format(len(context)))
+    log.info("Context: \033[91m {} \033[0m".format(context))
+    return context
+
+
+def main():
+    log.basicConfig(format="[ %(levelname)s ] %(message)s", level=log.INFO, stream=sys.stdout)
+    args = build_argparser().parse_args()
+
+    # load vocabulary file for model
+    log.info("Loading vocab file:\t{}".format(args.vocab))
+    with open(args.vocab, "r", encoding="utf-8") as r:
+        vocab = dict((t.rstrip("\n"), i) for i, t in enumerate(r.readlines()))
+    log.info("{} tokens loaded".format(len(vocab)))
+
+    # get context as a string (as we might need it's length for the sequence reshape)
+    context = get_context(args)
+    # encode context into token ids list
+    c_tokens_id, c_tokens_se = text_to_tokens(context, vocab)
+
+    log.info("Initializing Inference Engine")
+    ie = IECore()
+    log.info("Device is {}".format(args.device))
+    version = ie.get_versions(args.device)[args.device]
+    version_str = "{}.{}.{}".format(version.major, version.minor, version.build_number)
+    log.info("Plugin version is {}".format(version_str))
+
+    # read IR
+    model_xml = args.model
+    model_bin = os.path.splitext(model_xml)[0] + ".bin"
+    log.info("Loading network files:\n\t{}\n\t{}".format(model_xml, model_bin))
+    ie_encoder = ie.read_network(model=model_xml, weights=model_bin)
+
+    if args.reshape:
+        # reshape the sequence length to the context + maximum question length (in tokens)
+        first_input_layer = next(iter(ie_encoder.inputs))
+        c = ie_encoder.inputs[first_input_layer].shape[1]
+        # find the closest multiple of 64
+        seq = min(c, round((len(c_tokens_id) + args.max_question_token_num) / 64) * 64)
+        if seq < c:
```

Comment:
Why is only narrowing of input sequence allowed? Can't the net be increased in input size in the same way?

Reply:
It potentially can (up to the maximum sequence length that the network was trained for, usually 512).
But it brings no performance upside when the sequence are small (which is exactly the case we atrying to cover with reshaping here, to save on calculations).

Reply:
I agree that it's better to shrink the model when we don't need its full length.

But what prevents from exceeding the source length of 512? Is that correct that it is possible to decrease the shape but not increase? Usually it works both ways.

Reply:
It is possible to increase (to make less inference calls on the large contexts that do not fit the original sequence length), but this is not covered by this simple demo.
the limitation of the 512 comes from the fact that certain layers in the model are trained to see only 512 tokens at most.

Reply:
I think just having multiplied `c` in `seq = min(c, round((len(c_tokens_id) + args.max_question_token_num) / 64) * 64)` will cover that. With may be an inplace comment added to clarify that in general it is limited only by memory size on a device.

I also just noticed, should it be `(len(c_tokens_id) + args.max_question_token_num + 63) / 64) * 64` to round up instead of `round((len(c_tokens_id) + args.max_question_token_num) / 64) * 64`?

Reply:
So what about support of reshaping up? The rest looks fine.

Reply:
I have tested the reshaping up and it brings no performance value, but does introduce the accuracy issues, probably because there are some layers that accommodated to "see" only the sequence that the model was trained with. 
I discussed this with data scientists and they said are not aware on "upsampling" the language models.

Reply:
Ok, thank you!